lib/assets/javascripts/utils/validation.js

```diff
@@ -154,21 +154,30 @@ const getValidationMessage = (el) => {
   return getDefaultValidationMessage(getValidationTypeForElement(el));
 };
 const valid = (el) => {
-  $(el).parent().removeClass(validationStates.hasError);
-  $(el).attr(ariaInvalid(false));
-  $(el).next().hide();
+  const target = $(el);
+  const helpBlock = target.closest('form').find(`#${target.attr('aria-describedby')}`);
+  target.parent().removeClass(validationStates.hasError);
+  target.attr(ariaInvalid(false));
+  helpBlock.hide();
 };
 const invalid = (el) => {
-  $(el).parent().addClass(validationStates.hasError);
-  $(el).attr(ariaInvalid(true));
-  $(el).next().show();
+  const target = $(el);
+  const helpBlock = target.closest('form').find(`#${target.attr('aria-describedby')}`);
+  target.parent().addClass(validationStates.hasError);
+  target.attr(ariaInvalid(true));
+  helpBlock.show();
 };
 const addValidationMessage = (el, excludeAsterisks) => {
   const target = $(el);
   if (isString(target.attr('id'))) {
     const id = target.attr('id');
+    const typ = getValidationTypeForElement(target);
     if (!isString(target.attr('aria-describedby'))) {
-      target.after(blockHelp(`help-${id}`, getValidationMessage(el)));
+      if (typ === 'radio' || typ === 'cehckbox') {
```

Comment:
spelling typo at 'checkbox'. You have made changes in both validation logic JS which are used across every form in the site. Any change in this file should be cross-tested.

Reply:
thanks for catching this!